compiler/rustc_mir_build/src/build/mod.rs

```diff
@@ -837,7 +842,9 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             self.fn_span,
             self.coroutine,
             None,
-        )
+        );
+        body.coverage_hir_branch_info = self.coverage_branch_info.and_then(|b| b.into_done());
```

Comment:
Here I had a choice between two ways to set this field:
- Add an extra parameter to `Body::new`, and `None` to a half-dozen callers that don't care.
- Leave the constructor as-is, and do an explicit field write in the one caller that does care.

I think I prefer the explicit write, as shown here.